pkg/apiserver/rest/apis/v1/types.go

```diff
@@ -0,0 +1,250 @@
+	CPUCapacity      int64    `json:"cpuCapacity"`
+	GPUCapacity      int64    `json:"gpuCapacity,omitempty"`
+	StorageClassList []string `json:"storageClassList,omitempty"`
+}
+
+// ListClusterResponse list cluster
+type ListClusterResponse struct {
+	Clusters []ClusterBase `json:"clusters"`
+}
+
+// ClusterBase cluster base model
+type ClusterBase struct {
+	Name        string            `json:"name"`
+	Description string            `json:"description"`
+	Icon        string            `json:"icon"`
+	Labels      map[string]string `json:"labels"`
+	Status      string            `json:"status"`
+	Reason      string            `json:"reason"`
+}
+
+// ListClusterAddonResponse list all addon of the cluster
+type ListClusterAddonResponse struct {
+	Addons []ClusterAddonBase `json:"addons"`
+}
+
+// ClusterAddonBase cluster addon base model
+type ClusterAddonBase struct {
+	Name        string `json:"name"`
+	Description string `json:"description"`
+	Status      string `json:"status"`
+}
+
+// DeatilClusterAddonResponse detail addon info
+type DeatilClusterAddonResponse struct {
+	ClusterAddonBase
+}
+
+// ListApplicationResponse list applications by query params
+type ListApplicationResponse struct {
+	Applications []*ApplicationBase `json:"applications"`
+}
+
+// ApplicationBase application base model
+type ApplicationBase struct {
+	Name            string            `json:"name"`
+	Namespace       string            `json:"namespace"`
+	Description     string            `json:"description"`
+	CreateTime      time.Time         `json:"createTime"`
+	UpdateTime      time.Time         `json:"updateTime"`
+	Icon            string            `json:"icon"`
+	Labels          map[string]string `json:"labels,omitempty"`
+	ClusterBindList []ClusterBase     `json:"clusterList,omitempty"`
+	Status          string            `json:"status"`
+	GatewayRuleList []GatewayRule     `json:"gatewayRule"`
+}
+
+// RuleType gateway rule type
+type RuleType string
+
+const (
+	// HTTPRule Layer 7 HTTP policy.
+	HTTPRule RuleType = "http"
+	// StreamRule Layer 4 policy, such as TCP and UDP
+	StreamRule RuleType = "stream"
+)
+
+// GatewayRule application gateway rule
+type GatewayRule struct {
+	RuleType      RuleType `json:"ruleType"`
+	Address       string   `json:"address"`
+	Protocol      string   `json:"protocol"`
+	ComponentName string   `json:"componentName"`
+	ComponentPort int32    `json:"componentPort"`
+}
+
+// CreateApplicationRequest create application request body
+type CreateApplicationRequest struct {
+	Name        string            `json:"name" validate:"required"`
+	Namespace   string            `json:"namespace" validate:"required"`
+	Description string            `json:"description"`
+	Icon        string            `json:"icon"`
+	Labels      map[string]string `json:"labels,omitempty"`
+	ClusterList []string          `json:"clusterList,omitempty"`
+	YamlConfig  string            `json:"yamlConfig,omitempty"`
```

Comment:
why use yaml instead of using the Application Object?

Reply:
The scenario here is that the user uploads the YAML configuration for application creation.